src/routes/transactions/mappers/common/swap-order.mapper.ts

```diff
@@ -0,0 +1,239 @@
+          address: order.sellToken,
+        }),
+      ]);
+
+      const buyTokenAmount = new TokenAmount({
+        token: buyToken,
+        amount: order.buyAmount,
+        executedAmount: order.executedBuyAmount,
+      });
+      const sellTokenAmount = new TokenAmount({
+        token: sellToken,
+        amount: order.sellAmount,
+        executedAmount: order.executedSellAmount,
+      });
+
+      switch (order.status) {
+        case 'fulfilled':
+          return this._mapFulfilledOrderStatus({
+            buyToken: buyTokenAmount,
+            sellToken: sellTokenAmount,
+            order,
+          });
+        case 'open':
+        case 'cancelled':
+        case 'expired':
+          return this._mapDefaultOrderStatus({
+            buyToken: buyTokenAmount,
+            sellToken: sellTokenAmount,
+            order,
+          });
+        default:
+          return this._mapUnknownOrderStatus(chainId, transaction, dataSize);
+      }
+    } catch (e) {
+      this.loggingService.warn(e);
+      return this.customTransactionMapper.mapCustomTransaction(
+        transaction,
+        dataSize,
+        chainId,
+        null,
+        null,
+      );
+    }
+  }
+
+  private _mapUnknownOrderStatus(
+    chainId: string,
+    transaction: MultisigTransaction | ModuleTransaction,
+    dataSize: number,
+  ): Promise<CustomTransactionInfo> {
+    return this.customTransactionMapper.mapCustomTransaction(
+      transaction,
+      dataSize,
+      chainId,
+      null,
+      null,
+    );
+  }
+
+  private _getExecutionPrice(
+    sellToken: TokenAmount,
+    buyToken: TokenAmount,
+  ): string {
+    const ratio = sellToken.getExecutedAmount() / buyToken.getExecutedAmount();
+    return `1 ${sellToken.token.symbol} = ${ratio} ${buyToken.token.symbol}`;
+  }
+
+  private _getLimitPrice(
+    sellToken: TokenAmount,
+    buyToken: TokenAmount,
+  ): string {
+    const ratio = sellToken.getAmount() / buyToken.getAmount();
+    return `1 ${sellToken.token.symbol} = ${ratio} ${buyToken.token.symbol}`;
+  }
+
+  private _mapFulfilledOrderStatus(args: {
+    buyToken: TokenAmount;
+    sellToken: TokenAmount;
+    order: Order;
+  }): SwapOrderTransactionInfo {
+    if (args.order.kind === 'unknown') {
+      throw new Error('Unknown order kind');
+    }
+    const surplusFee: string | null = args.order.executedSurplusFee
+      ? this._getExecutedSurplusFee(
+          args.order.executedSurplusFee,
+          args.buyToken.token,
+        )
+      : null;
+
+    return new FulfilledSwapOrderTransactionInfo({
+      orderKind: args.order.kind,
+      sellToken: args.sellToken.toTokenInfo(),
+      buyToken: args.buyToken.toTokenInfo(),
+      expiresTimestamp: args.order.validTo,
+      surplus: surplusFee,
+      executionPrice: this._getExecutionPrice(args.sellToken, args.buyToken),
+    });
+  }
+
+  private _getExecutedSurplusFee(
+    executedSurplusFee: bigint,
+    token: Token & { decimals: number },
+  ): string | null {
```

Comment:
```suggestion
  ): string {
```

Reply:
Changed in [e3a4a9c](https://github.com/safe-global/safe-client-gateway/pull/1299/commits/e3a4a9c33aac2d31572dd3cb51e6cacec9ed3515)